### PR TITLE
 added side-effect, custom hook and fragment

### PR DESCRIPTION
**Pull request by Mariela-t**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,20 +1,33 @@
...
+  const [todoList, setTodoList] = useState(
+    JSON.parse(localStorage.getItem("savedTodoList")) || []
+  );
+
+  useEffect(() => {
+    localStorage.setItem("savedTodoList", JSON.stringify(todoList));
+  }, [todoList]);
+
+  return [todoList, setTodoList];
+};
```
Comment (kwilson05) [inline src/App.js:15]: 💯 Looking good!!

File: src/App.js [hunk 2]
```diff
@@ -1,20 +1,33 @@
...
+
 function App() {
-  const [todoList, setTodoList] = useState([]);
+  const [todoList, setTodoList] = useSemiPersistentState();
 
   const addTodo = (newTodo) => {
     setTodoList([...todoList, newTodo]);
   };
```
Comment (kwilson05) [inline src/App.js:22]: So this is my fault -- After more research, I found the way we set the new array was wrong. React allows us to provide a function to the setter (setTodoList) and function param will get the most recent value of the state.

So it'd look something like this:

```suggestion
  const addTodo = (newTodo) => {
     setTodoList( (previousTodoList) => [...previousTodoList, newTodo])
  };
```
I used an arrow function there b/c it's easier to read. How does all this sound to you?

Exercise 1.4 overview section actually explains my comment a bit better 😆
https://github.com/Code-the-Dream-School/react-cockatoo/wiki/Lesson-1.4
take a look

**Review by kwilson05 — APPROVED** [review]
Everything looks good here, Mariela 🍀 

I made one comment but you can correct that while working on exercise 1.6 :)

File: src/App.js (hunk 2 above)
Comment (Mariela-t) [reply src/App.js:22]: Is this because we pass the copy of the array or object into the setter function?

Comment (kwilson05) [reply src/App.js:22]: Yeah! Its b/c we depend on the old state to create a new one.

Comment (Mariela-t) [reply src/App.js:22]: Okay, Thank you!🙂

**Pull request merged**